arcsi/api/item.py

```diff
@@ -422,17 +482,234 @@ def archon_edit_item(id):
             .all()
         )
         item.tags = (
-            get_or_create(Tag, display_name=tag.display_name, clean_name=normalise(tag.display_name)) for tag in item_metadata.tags
+            get_or_create(
+                Tag,
+                display_name=tag.display_name,
+                clean_name=normalise(tag.display_name),
+            )
+            for tag in item_metadata.tags
         )
-        
+        # TODO Implement upsert https://docs.sqlalchemy.org/en/20/dialects/postgresql.html#insert-on-conflict-upsert
         db.session.add(item)
         db.session.flush()
+        empty_files = all([not (_v) for _v in request.files.values()])
+        request_files = {k: v for k, v in request.files.items()}
```

Comment:
In my PR I totally missed this part where you separated the handling of the form depending on the existence of the attached image and audio files.
This could be a nice potential addition in the future (or maybe we can handle this under your media schema idea)